diff --git a/node_modules/@react-three/fiber/dist/events-11ca7a92.cjs.prod.js b/node_modules/@react-three/fiber/dist/events-11ca7a92.cjs.prod.js
index 7ee6974..effd58c 100644
--- a/node_modules/@react-three/fiber/dist/events-11ca7a92.cjs.prod.js
+++ b/node_modules/@react-three/fiber/dist/events-11ca7a92.cjs.prod.js
@@ -351,7 +351,7 @@ function diffProps(instance, newProps) {
   // Sort through props
   for (const prop in newProps) {
     // Skip reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     // Skip if props match
     if (is.equ(newProps[prop], instance.props[prop])) continue;
 
@@ -366,7 +366,7 @@ function diffProps(instance, newProps) {
 
   // Reset removed props for HMR
   for (const prop in instance.props) {
-    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     const {
       root,
       key
@@ -403,7 +403,7 @@ function applyProps(object, props) {
     let value = props[prop];
 
     // Don't mutate reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
 
     // Deal with pointer events, including removing them if undefined
     if (instance && EVENT_REGEX.test(prop)) {
diff --git a/node_modules/@react-three/fiber/dist/events-508aad4b.cjs.dev.js b/node_modules/@react-three/fiber/dist/events-508aad4b.cjs.dev.js
index c46d5e6..469e792 100644
--- a/node_modules/@react-three/fiber/dist/events-508aad4b.cjs.dev.js
+++ b/node_modules/@react-three/fiber/dist/events-508aad4b.cjs.dev.js
@@ -351,7 +351,7 @@ function diffProps(instance, newProps) {
   // Sort through props
   for (const prop in newProps) {
     // Skip reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     // Skip if props match
     if (is.equ(newProps[prop], instance.props[prop])) continue;
 
@@ -366,7 +366,7 @@ function diffProps(instance, newProps) {
 
   // Reset removed props for HMR
   for (const prop in instance.props) {
-    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     const {
       root,
       key
@@ -403,7 +403,7 @@ function applyProps(object, props) {
     let value = props[prop];
 
     // Don't mutate reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
 
     // Deal with pointer events, including removing them if undefined
     if (instance && EVENT_REGEX.test(prop)) {
diff --git a/node_modules/@react-three/fiber/dist/events-760a1017.esm.js b/node_modules/@react-three/fiber/dist/events-760a1017.esm.js
index f4e89a7..d3ca918 100644
--- a/node_modules/@react-three/fiber/dist/events-760a1017.esm.js
+++ b/node_modules/@react-three/fiber/dist/events-760a1017.esm.js
@@ -326,7 +326,7 @@ function diffProps(instance, newProps) {
   // Sort through props
   for (const prop in newProps) {
     // Skip reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     // Skip if props match
     if (is.equ(newProps[prop], instance.props[prop])) continue;
 
@@ -341,7 +341,7 @@ function diffProps(instance, newProps) {
 
   // Reset removed props for HMR
   for (const prop in instance.props) {
-    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || newProps.hasOwnProperty(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
     const {
       root,
       key
@@ -378,7 +378,7 @@ function applyProps(object, props) {
     let value = props[prop];
 
     // Don't mutate reserved keys
-    if (RESERVED_PROPS.includes(prop)) continue;
+    if (RESERVED_PROPS.includes(prop) || (typeof prop === 'string' && prop.startsWith('x-'))) continue;
 
     // Deal with pointer events, including removing them if undefined
     if (instance && EVENT_REGEX.test(prop)) {
